### ISSUE TITLE
[operator-sdk] User "system:serviceaccount:default:nginx-operator" cannot get resource "pods" in API group "" in the namespace "default"

### ISSUE THREAD

**Issue by Jinchul81** [operator-framework/operator-sdk#1491]
## Bug Report

**What did you do?**
I have followed up the guideline of Helm operator-sdk in [user-guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/helm/user-guide.md)

The unexpected error happened at the end of the "Run as a pod inside a Kubernetes cluster"
Here are my command lists:
```
operator-sdk new nginx-operator --api-version=example.com/v1alpha1 --kind=Nginx --type=helm
cd nginx-operator
kubectl create -f deploy/crds/example_v1alpha1_nginx_crd.yaml
operator-sdk build myaccount/nginx-operator:v0.0.1
docker push myaccount/nginx-operator:v0.0.1
sed -i 's|REPLACE_IMAGE|myaccount/nginx-operator:v0.0.1|g' deploy/operator.yaml
kubectl create -f deploy/service_account.yaml
kubectl create -f deploy/role.yaml
kubectl create -f deploy/role_binding.yaml
kubectl create -f deploy/operator.yaml
```

**What did you expect to see?**
No error from the pod state

**What did you see instead? Under which circumstances?**
> User "system:serviceaccount:default:nginx-operator" cannot get resource "pods" in API group "" in the namespace "default"

Here are the detailed logs:
```
$kubectl get pods
NAME                              READY   STATUS             RESTARTS   AGE
nginx-operator-5c6946d8d4-vwgnb   0/1     CrashLoopBackOff   1          12s

$ kubectl logs nginx-operator-5c89cffb5b-ffnfs
{"level":"info","ts":1559050310.9451957,"logger":"cmd","msg":"Go Version: go1.12.5"}
{"level":"info","ts":1559050310.945242,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1559050310.9452474,"logger":"cmd","msg":"Version of operator-sdk: v0.8.0"}
{"level":"info","ts":1559050310.9452708,"logger":"cmd","msg":"Watching single namespace.","Namespace":"default"}
{"level":"info","ts":1559050311.080124,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"nginx-controller","source":"kind source: example.com/v1alpha1, Kind=Nginx"}
{"level":"info","ts":1559050311.0803094,"logger":"helm.controller","msg":"Watching resource","apiVersion":"example.com/v1alpha1","kind":"Nginx","namespace":"default","reconcilePeriod":"1m0s"}
{"level":"info","ts":1559050311.080329,"logger":"leader","msg":"Trying to become the leader."}
{"level":"error","ts":1559050311.169351,"logger":"k8sutil","msg":"Failed to get Pod","Pod.Namespace":"default","Pod.Name":"nginx-operator-5c89cffb5b-ffnfs","error":"pods \"nginx-operator-5c89cffb5b-ffnfs\" is forbidden: User \"system:serviceaccount:default:nginx-operator\" cannot get resour
ce \"pods\" in API group \"\" in the namespace \"default\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/
operator-sdk/pkg/k8sutil.GetPod\n\tsrc/github.com/operator-framework/operator-sdk/pkg/k8sutil/k8sutil.go:105\ngithub.com/operator-framework/operator-sdk/pkg/leader.myOwnerRef\n\tsrc/github.com/operator-framework/operator-sdk/pkg/leader/leader.go:142\ngithub.com/operator-framework/operator-s
dk/pkg/leader.Become\n\tsrc/github.com/operator-framework/operator-sdk/pkg/leader/leader.go:67\ngithub.com/operator-framework/operator-sdk/pkg/helm.Run\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/run.go:104\ngithub.com/operator-framework/operator-sdk/cmd/operator-sdk/run.newR
unHelmCmd.func1\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/run/helm.go:38\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).execute\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:762\ngith
ub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).Execute\n\tsrc/github.
com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/main.go:80\nruntime.main\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/runtime/proc.go:200"}
{"level":"error","ts":1559050311.1694436,"logger":"cmd","msg":"Failed to become leader.","Namespace":"default","error":"pods \"nginx-operator-5c89cffb5b-ffnfs\" is forbidden: User \"system:serviceaccount:default:nginx-operator\" cannot get resource \"pods\" in API group \"\" in the namespace \"default\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm.Run\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/run.go:106\ngithub.com/operator-framework/operator-sdk/cmd/operator-sdk/run.newRunHelmCmd.func1\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/run/helm.go:38\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra
.(*Command).execute\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/co
mmand.go:852\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).Execute\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/main.go:80\n
runtime.main\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/runtime/proc.go:200"}
Error: pods "nginx-operator-5c89cffb5b-ffnfs" is forbidden: User "system:serviceaccount:default:nginx-operator" cannot get resource "pods" in API group "" in the namespace "default"
Usage:
  operator-sdk run helm [flags]

Flags:
  -h, --help                        help for helm
      --reconcile-period duration   Default reconcile period for controllers (default 1m0s)
      --watches-file string         Path to the watches file to use (default "./watches.yaml")
      --zap-devel                   Enable zap development mode (changes defaults to console encoder, debug log level, and disables sampling)
      --zap-encoder encoder         Zap log encoding ('json' or 'console')
      --zap-level level             Zap log level (one of 'debug', 'info', 'error' or any integer value > 0) (default info)
      --zap-sample sample           Enable zap log sampling. Sampling will be disabled for integer log levels > 1

Global Flags:
      --verbose   Enable verbose logging
```

**Environment**
* operator-sdk version:

> operator-sdk version: v0.8.0, commit: 78c472461e75e6c64589cfadf577a2004b8a26b3

* go version:

go version go1.12.5 linux/amd64

* Kubernetes version information:

> Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
> Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.6", GitCommit:"ab91afd7062d4240e95e51ac00a18bd58fddd365", GitTreeState:"clean", BuildDate:"2019-02-26T12:49:28Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

* Are you writing your operator in ansible, helm, or go?
I was just using the example in the user guide. There wasn't any change.

**Possible Solution**

**Additional context**

**Comment by joelanford** [comment]
@Jinchul81 Thanks for the report. It looks like the RBAC generation for helm is not including the necessary rule for the leader election feature to work. As a workaround until we get this fixed, you should be able to add the following to `deploy/role.yaml` to resolve the issue.

```yaml
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - get
```

**Comment by joelanford** [comment]
Actually, I'm fairly confident this is a bug that was fixed in `v0.8.1` (see https://github.com/operator-framework/operator-sdk/pull/1456 for details). Can you upgrade to `v0.8.1` and retry? If upgrading to `v0.8.1` does not resolve your issue please re-open and we can take a closer look.

**Comment by Jinchul81** [comment]
@joelanford Thanks a lot for your quick answer! I confirmed that `v0.8.1` should resolve the issue!
